src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -494,8 +505,59 @@ private void requestMaterials()
         final JobBuilder builderJob = (JobBuilder) job;
         while (builderJob.getStructure().findNextBlock())
         {
-            @Nullable final Block block = builderJob.getStructure().getBlock();
-            @NotNull final IBlockState blockState = builderJob.getStructure().getBlockState();
+            @Nullable final Template.BlockInfo blockInfo = builderJob.getStructure().getBlockInfo();
+            @Nullable final Template.EntityInfo entityInfo = builderJob.getStructure().getEntityinfo();
+
+            if(blockInfo == null)
+            {
+                continue;
+            }
+
+            @Nullable final IBlockState blockState = blockInfo.blockState;
+            @Nullable final Block block = blockState.getBlock();
+
+            if (entityInfo != null)
+            {
+                Entity entity;
+                try
+                {
+                    entity = EntityList.createEntityFromNBT(entityInfo.entityData, world);
+                }
+                catch (Exception e)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Catch a list of specific exception subtypes instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2221)